src/scheduler.rs

```diff
@@ -89,21 +100,39 @@ impl ScheduledPayJoin {
             .iter()
             .map(|channel| channel.amount)
             .fold(bitcoin::Amount::ZERO, std::ops::Add::add);
-        // TODO: replace with sheduled_payjoin.fees()
-        let fees = calculate_fees(
-            self.channels.len() as u64,
-            self.fee_rate,
-            self.wallet_amount() != bitcoin::Amount::ZERO,
-        );
-        let wallet_amount = self.wallet_amount();
+
+        let reserve_deposit = self.reserve_deposit();
 
         let owned_txout_value = our_output.value;
 
-        (total_channel_amount + fees + wallet_amount).as_sat() == owned_txout_value
+        (total_channel_amount + self.fees() + reserve_deposit).as_sat() == owned_txout_value
     }
 
-    /// This externally exposes [ScheduledPayJoin]::wallet_amount.
-    pub fn wallet_amount(&self) -> bitcoin::Amount { self.wallet_amount }
+    /// This externally exposes [ScheduledPayJoin]::reserve_deposit.
+    pub fn reserve_deposit(&self) -> bitcoin::Amount { self.reserve_deposit }
+
+    /// Calculate the absolute miner fee this [ScheduledPayJoin] pays
+    fn fees(&self) -> bitcoin::Amount {
+        let channel_count = self.channels.len() as u64;
+        let has_reserve_deposit = self.reserve_deposit != bitcoin::Amount::ZERO;
+
+        let mut additional_vsize = if has_reserve_deposit {
+            // <8 invariant bytes = 4 version + 4 locktime>
+            //  + 2 variant bytes for input.len + output.len such that each len < 252
+            //  + OP_0 OP_PUSHBYTES_32 <32 byte script>
+            channel_count * (8 + 1 + 1 + 34)
+        } else {
+            // substitute 1 p2wsh channel (34 bytes) open for 1 p2wpkh reserve output (22 bytes)
+            // that's + 12 bytes
+            (channel_count - 1) * (8 + 1 + 1 + 34) + 12
+        };
+
+        if self.quote.is_some() {
+            additional_vsize = additional_vsize + (8 + 1 + 1 + 22); // P2WPKH (OP_0 OP_PUSHBYTES_20 <20 byte script)
+        }
+
+        bitcoin::Amount::from_sat(self.fee_rate * additional_vsize)
```

Comment:
I think this might slightly overestimate the fees. I think that's acceptable to me for an alpha. This fee calculation belongs in the payjoin crate and should be perfect for any script type

Reply:
I reckon we should try avoid doing diy fees, is there a reason we're having to calculate yourself as opposed to rust-bitcoin or something?

Reply:
Yes, we should avoid DIY fee calc. It can lead to many problems if done poorly.

I don't believe rust-bitcoin and rust-miniscript have any fee calculation. BDK is in its infancy for fee calculation.

State of the art fee calculation candidate for rust-bitcoin is actually in the payjoin crate but only exposed to the sender module. You can follow the [proposal to upstream the relevant code in rust-bitcoin](https://github.com/rust-bitcoin/rust-bitcoin/issues/630) here  and the (good but more testing would be better) [fee calculation in the payjoin crate here](https://github.com/Kixunil/payjoin/blob/150dd33b9c60a4e1c219955b9af2e5aca0a43bec/bip78/src/sender/mod.rs#L196-L225)

Reply:
see https://github.com/rust-bitcoin/rust-bitcoin/pull/1338/commits/c34d5f8f859c151df77bbdef7e0ef212a80ed69d recently in master but not yet in a release

Reply:
The important thing to remember is the sender always signs off on the fee calculation. This just sets a minimum.